### PR TITLE
[ModelZoo] Refactor ERNIE-Layout Usage and add unittest

### PR DESCRIPTION
**Pull request by linjieccc**
### PR types

Function optimization
### PR changes

APIs
### Description

- Refactor ERNIE-Layout Usage
- Add unittest for ERNIE-Layout

### CODE REVIEW

File: paddlenlp/transformers/ernie_layout/configuration.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    >>> from paddlenlp.transformers import ErnieLayoutModel, ErnieLayoutConfig
+    >>> # Initializing a ErnieLayout ernie-layoutx-base-uncased configuration
+    >>> configuration = ErnieLayoutConfig()
+    >>> # Initializing a model from the  style configuration
+    >>> model = ErnieLayoutModel(configuration)
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "ernie_layout"
+    attribute_map: Dict[str, str] = {"num_classes": "num_labels", "dropout": "classifier_dropout"}
```
Comment (sijunhe) [inline paddlenlp/transformers/ernie_layout/configuration.py:153]: 这个会直接继承，不用加

File: paddlenlp/transformers/__init__.py [hunk 2]
```diff
@@ -75,6 +75,7 @@
 from .ernie_gram.tokenizer import *
 from .ernie_layout.modeling import *
 from .ernie_layout.tokenizer import *
+from .ernie_layout.configuration import *
```
Comment (sijunhe) [inline paddlenlp/transformers/__init__.py:80]: 顺序应该是
```
from .ernie_layout.configuration import *
from .ernie_layout.modeling import *
from .ernie_layout.tokenizer import *
```

File: paddlenlp/transformers/ernie_layout/modeling.py [hunk 3]
```diff
@@ -1121,16 +1057,16 @@ def forward(
...
+    def __init__(self, config: ErnieLayoutConfig):
+        super(ErnieLayoutForQuestionAnswering, self).__init__(config)
+        self.num_labels = config.num_labels
+        self.ernie_layout = ErnieLayoutModel(config)
+        self.has_visual_segment_embedding = config.has_visual_segment_embedding
+        classifier_dropout = (
+            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
+        )
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.qa_outputs = nn.Linear(config["hidden_size"], config.num_labels)
```
Comment (sijunhe) [inline paddlenlp/transformers/ernie_layout/modeling.py]:
```suggestion
        self.qa_outputs = nn.Linear(config["hidden_size"], 2)
```

File: tests/transformers/ernie_layout/test_modeling.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+            input_ids,
+            position_ids=position_ids,
+            attention_mask=attention_mask,
+            bbox=bbox,
+            image=image,
+            start_positions=sequence_labels,
+            end_positions=sequence_labels,
+        )
+
+        start_logits, end_logits = result[0]
```
Comment (sijunhe) [inline tests/transformers/ernie_layout/test_modeling.py]:
```suggestion
        if sequence_labels is not None:
            start_logits, end_logits = result[1], result[2]
        else:
            start_logits, end_logits = result[0], result[1]
```

File: tests/transformers/ernie_layout/test_modeling.py [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+
+        self.parent.assertEqual(start_logits.shape, [self.batch_size, self.seq_length, self.num_classes])
+        self.parent.assertEqual(end_logits.shape, [self.batch_size, self.seq_length, self.num_classes])
```
Comment (sijunhe) [inline tests/transformers/ernie_layout/test_modeling.py]:
```suggestion
        self.parent.assertEqual(start_logits.shape, [self.batch_size, self.seq_length])
        self.parent.assertEqual(end_logits.shape, [self.batch_size, self.seq_length])
```

File: paddlenlp/transformers/ernie_layout/configuration.py (hunk 1 above)
Comment (sijunhe) [inline paddlenlp/transformers/ernie_layout/configuration.py:153]:
```suggestion
```

File: paddlenlp/transformers/ernie_layout/tokenizer.py [hunk 6]
```diff
@@ -74,6 +75,8 @@ def __init__(
         self,
         vocab_file,
         sentencepiece_model_file,
+        do_lower_case=True,
+        encoding="utf8",
```
Comment (sijunhe) [inline paddlenlp/transformers/ernie_layout/tokenizer.py]:
```suggestion
```

File: paddlenlp/transformers/ernie_layout/tokenizer.py [hunk 7]
```diff
@@ -88,7 +91,10 @@ def __init__(
         self._unk_token = unk_token
         self._pad_token = pad_token
         self._mask_token = mask_token
+        self.do_lower_case = do_lower_case
+        self.encoding = encoding
         self.sp_model = spm.SentencePieceProcessor()
+        self.vocab = self.load_vocabulary(vocab_file, unk_token=unk_token)
```
Comment (sijunhe) [inline paddlenlp/transformers/ernie_layout/tokenizer.py]:
```suggestion
        self.sp_model = spm.SentencePieceProcessor()
```

**Review by sijunhe — APPROVED** [review]
merging for now. We will leave tokenizer test for the next PR
